openpype/hosts/substancepainter/api/lib.py

```diff
@@ -588,6 +588,7 @@ def _setup_file_dialog():
         # the file
         while not file_dialog.selectedFiles():
             app.processEvents(QtCore.QEventLoop.ExcludeUserInputEvents, 1000)
+            continue
```

Comment:
```python

        directory = os.path.dirname(mesh_filepath)
        url = QtCore.QUrl.fromLocalFile(os.path.basename(mesh_filepath))
        
        file_dialog.setDirectory(directory)
        file_dialog.selectUrl(url)

        # Give the explorer window time to refresh to the folder and select
        # the file
        start = time.time()
        while not file_dialog.selectedFiles() and time.time() < start + 10000:  # timeout 10 seconds
            app.processEvents(QtCore.QEventLoop.ExcludeUserInputEvents, 2000)
            # Continuously try to select the file, sometimes it just doesn't
            # want to trigger the select if the set directory call from before
            # was still initializing the folder
            file_dialog.selectUrl(url)
            
        if not file_dialog.selectedFiles():
            print("Unable to select the file in the dialog. Unexpected behavior may occur.")
        else:
            print(f"Selected: {file_dialog.selectedFiles()}")

        # Set it again now we know the path is refreshed - without this
        # accepting the dialog will often not trigger the correct filepath
        file_dialog.setDirectory(url))
        file_dialog.selectUrl(url)
```

@moonyuet you could try something like this? This way it tries to select the file over time again and again.

Reply:
it is still struck at create project dialog.
https://github.com/ynput/OpenPype/assets/64118225/db613f36-6e5a-4118-8f33-5e427f16c54f

Reply:
Interesting - it appears actually to hang elsewhere since it looks like it's already set the filepath (e.g. shows it at the top of the dialog) but something is holding the dialog from being able to "accept" (not also how "ok" button is greyed out)

What happens if you open that dialog manually (new project?) and then set the file manually, can you directly click OK or do you need to configure something else, if so; what?

Reply:
> Interesting - it appears actually to hang elsewhere since it looks like it's already set the filepath (e.g. shows it at the top of the dialog) but something is holding the dialog from being able to "accept" (not also how "ok" button is greyed out)
> 
> What happens if you open that dialog manually (new project?) and then set the file manually, can you directly click OK or do you need to configure something else, if so; what?

When I set it manually, it is usually just select the filepath of the mesh and it will show up and then click **Ok**
![image](https://github.com/ynput/OpenPype/assets/64118225/bb3f905b-a467-4fb3-9e0c-63d20b889373)